dsew_community_profile/tests/test_pull.py

```diff
@@ -421,3 +434,37 @@ def test_std_err(self):
                     "sample_size": [2, 2, 5, 10, 20, 0]
                 })
             )
+
+    def test_interpolation(self):
+        ex1 = Example(
+            geo_id = ["1"] * 8 + ["2"] * 6,
+            timestamp = list(pd.date_range("2022-01-01", "2022-01-05")) + list(pd.date_range("2022-01-08", "2022-01-10")) + list(pd.date_range("2022-01-01", "2022-01-03")) + list(pd.date_range("2022-01-08", "2022-01-10")),
+            val = [i ** 2 - i + 5 for i in range(2, 12) if i % 7 >= 2] + [3 * i ** 2 + 4 * i - 10 for i in range(4, 14) if i % 7 >= 4]
```

Comment:
what does this do?

Reply:
so it's a test of a data signal with two different geos and with two different missing lengths (the first one is missing dates 2022-01-06 and 2022-01-07 and the second one misses 2022-01-04 -- 2022-01-7). the values of each signal comes from one of two second-degree polynomials, something that a cubic interpolator can infer with 100% accuracy.

Reply:
cool -- let's make that more legible; i got lost among the exponents and moduli. mix-n-match strategies for this could include:
* write a comment explaining what is being done and why ("something that a cubic interpolator can infer with 100% accuracy" is 💯  but non-obvious)
* define the second-degree polynomials first in a dedicated variable or function, then handle the gap logic when building the data frame
* define the data frame for the dense date range, then selectively drop the desired missing days